### PR TITLE
[LC-232] add get_citizens in channel_inner_service

### PR DESCRIPTION
**Pull request by laurenceyoon**
- add get_citizens in channel_inner_service
- fix confirm_info type as bytes, except for jsonRPC/websocket params.
- update subscribe_network on_enter in statemachine

### CODE REVIEW

File: loopchain/channel/channel_inner_service.py [hunk 1]
```diff
@@ -390,7 +391,7 @@ def __init__(self, channel_service: 'ChannelService'):
 
         # Citizen
         self._citizen_condition_new_block: Condition = None
-        self._citizen_set = set()
+        self._citizens = dict()  # {'peer_id': Citizen}
```
Comment (zsaladin) [inline loopchain/channel/channel_inner_service.py]: Could you use type hint?

```python
from typing import Dict

self._citizens: Dict[str, Citizen]
```

File: loopchain/channel/channel_inner_service.py [hunk 2]
```diff
@@ -490,37 +491,47 @@ def __add_tx_list(self, tx_list):
...
-    async def register_subscriber(self, peer_id):
-        if len(self._citizen_set) >= conf.SUBSCRIBE_LIMIT:
+    async def register_citizen(self, peer_id, target, connected_time):
+        if len(self._citizens) >= conf.SUBSCRIBE_LIMIT:
             return False
         else:
-            self._citizen_set.add(peer_id)
-            logging.info(f"register new subscriber: {peer_id}")
-            logging.debug(f"remaining all subscribers: {self._citizen_set}")
+            Citizen = namedtuple("Citizen", "peer_id target connected_time")
```
Comment (zsaladin) [inline loopchain/channel/channel_inner_service.py]: How about moving this Citizen declaration to top of this file to use it on type hint?
Ultimately, it should be placed in a separate location later.

File: loopchain/channel/channel_inner_service.py [hunk 3]
```diff
@@ -490,37 +491,47 @@ def __add_tx_list(self, tx_list):
...
+    @message_queue_task
+    async def is_citizen_registered(self, peer_id) -> bool:
+        return peer_id in self._citizens
 
     @message_queue_task
-    async def is_registered_subscriber(self, peer_id):
-        return peer_id in self._citizen_set
+    async def get_citizens(self) -> list:
+        return [{"id": ctz.peer_id, "target": ctz.target, "connected_time": ctz.connected_time}
+                for peer_id, ctz in self._citizens.items()]
```
Comment (zsaladin) [inline loopchain/channel/channel_inner_service.py]: `peer_id` does not needed.
self._citizens.values()

File: loopchain/channel/channel_inner_service.py (hunk 1 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_inner_service.py]: No prob! Thanks for suggestion.

File: loopchain/peer/block_manager.py [hunk 4]
```diff
@@ -397,7 +397,7 @@ def __block_request_by_citizen(self, block_height, rs_rest_stub):
         block_version = self.get_blockchain().block_versioner.get_version(block_height)
         block_serializer = BlockSerializer.new(block_version, self.get_blockchain().tx_versioner)
         block = block_serializer.deserialize(get_block_result['block'])
-        confirm_info = get_block_result['confirm_info'] if 'confirm_info' in get_block_result else None
+        confirm_info = get_block_result.get('confirm_info', '').encode('utf-8')
```
Comment (yakkle) [inline loopchain/peer/block_manager.py]: I think `confirm_info` already converted by `encode('utf-8')` your [PR](https://github.com/icon-project/icon-rpc-server/pull/61/files#diff-9761d03b1583578cd176e1cb0f2e5759R157) in icon-rpc-server.

File: loopchain/channel/channel_inner_service.py (hunk 2 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_inner_service.py]: CitizenInfo(namedtuple ) has been defined as member variable to use it inside methods and type hint. Thanks for your help:)

File: loopchain/peer/block_manager.py (hunk 4 above)
Comment (yakkle) [reply loopchain/peer/block_manager.py]: my misunderstanding

File: loopchain/channel/channel_inner_service.py [hunk 5]
```diff
@@ -490,37 +492,46 @@ def __add_tx_list(self, tx_list):
...
+            logging.warning(f"already unregistered citizen({peer_id})")
+
+    @message_queue_task
+    async def is_citizen_registered(self, peer_id) -> bool:
+        return peer_id in self._citizens
 
     @message_queue_task
-    async def is_registered_subscriber(self, peer_id):
-        return peer_id in self._citizen_set
+    async def get_citizens(self) -> list:
```
Comment (yakkle) [inline loopchain/channel/channel_inner_service.py]:
```suggestion
    async def get_citizens(self) -> List[Dict[str, str]]:
```

Comment (laurenceyoon) [reply loopchain/channel/channel_inner_service.py]: Got it!

**Review by zsaladin — APPROVED** [review]

File: loopchain/channel/channel_inner_service.py [hunk 6]
```diff
@@ -490,37 +492,46 @@ def __add_tx_list(self, tx_list):
 
             new_block_height = subscriber_block_height + 1
             new_block = blockchain.find_block_by_height(new_block_height)
-            confirm_info: str = blockchain.find_confirm_info_by_height(new_block_height).decode("utf-8")
+            confirm_info: bytes = blockchain.find_confirm_info_by_height(new_block_height)
```
Comment (yakkle) [inline loopchain/channel/channel_inner_service.py]: Is confirm_info `bytes` or `bytearray`?
I suggest to change `find_confirm_info_by_height` return bytes.

**Review by yakkle — APPROVED** [review]

**Comment by yakkle** [comment]
target branch changed.
Please rebase with develop.

File: loopchain/channel/channel_inner_service.py (hunk 6 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_inner_service.py]: Updated to bytes!

**Pull request merged**